### PR TITLE
US3632 - Optimization Link

### PR DESCRIPTION
**Pull request by RealNobody**
## CHANGES

Added classes for getting Optimization Tasks and Optimization Links.

When I finally started working on them, I realized that they should be separate objects, so I moved the YML around accordingly

## DEVELOPER NOTES

- [ ] Yext::Api::KnowledgeApi::OptimizationTasks::OptimizationTask
- [ ] Yext::Api::KnowledgeApi::OptimizationTasks::OptimizationLink

## TESTING NOTES

- [ ] List Optimization Tasks and the Locations that have completed or need to complete them.
- [ ] Get an Optimization Link for specific Locations and/or tasks.

### CODE REVIEW

File: spec/yext/api/knowledge_api/knowledge_manager/location_spec.rb [hunk 1]
```diff
@@ -7,13 +7,28 @@
   it_behaves_like "default scopes"
 
   describe "index" do
-    it_behaves_like "sets rate limits", -> { Yext::Api::KnowledgeApi::KnowledgeManager::Location.all.to_a }
+    it_behaves_like "sets rate limits", -> { Yext::Api::KnowledgeApi::KnowledgeManager::Location.account("my-test-account-id-2").all.to_a }
```
Comment (RealNobody) [inline spec/yext/api/knowledge_api/knowledge_manager/location_spec.rb]: I wasn't testing Locations properly before due to my not understanding them properly.

So, I fixed this test.

File: spec/yext/api/knowledge_api/optimization_tasks/optimization_link_spec.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+          get
+
+      expect(link.link).to be
+    end
+
+    it "gets a link for a specific task" do
+      link = Yext::Api::KnowledgeApi::OptimizationTasks::OptimizationLink.
+          account("my-test-account-id-2").
+          location("my-test-location-id-2").
+          task(1).
```
Comment (RealNobody) [inline spec/yext/api/knowledge_api/optimization_tasks/optimization_link_spec.rb]: Link to 1 task.

File: spec/yext/api/knowledge_api/optimization_tasks/optimization_link_spec.rb [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+          get
+
+      expect(link.link).to be
+    end
+
+    it "gets a link for a set of tasks" do
+      link = Yext::Api::KnowledgeApi::OptimizationTasks::OptimizationLink.
+          account("my-test-account-id-2").
+          location("my-test-location-id-2").
+          tasks(1, 2, 3, 4).
```
Comment (RealNobody) [inline spec/yext/api/knowledge_api/optimization_tasks/optimization_link_spec.rb]: Link to specific tasks

File: spec/yext/api/knowledge_api/optimization_tasks/optimization_link_spec.rb [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+          get
+
+      expect(link.link).to be
+    end
+
+    it "gets a link for a set of tasks with a particular mode" do
+      link = Yext::Api::KnowledgeApi::OptimizationTasks::OptimizationLink.
+          account("my-test-account-id-2").
+          location("my-test-location-id-2").
+          mode(Yext::Api::Enumerations::OptimizationLinkMode::PENDING_ONLY).
```
Comment (RealNobody) [inline spec/yext/api/knowledge_api/optimization_tasks/optimization_link_spec.rb]: You can get all, only pending tasks, or "reset" one or more tasks.

File: lib/yext/api/administrative_api/account.rb [hunk 5]
```diff
@@ -37,7 +37,7 @@ class Account < Yext::Api::Utils::ApiBase
         # Furthermore, the account_id cannot be changed by the partner, but only by the customer
         # themselves.
         def save
-          return unless accountId?
+          return unless Yext::Api::Validators::AccountValidator.new(self).valid?
```
Comment (RealNobody) [inline lib/yext/api/administrative_api/account.rb]: Previous PR feedback I hadn't had time to get to till now.

**Comment by krimsonkla** [comment]
👍

**Pull request merged**